pinot-query-planner/src/main/java/org/apache/pinot/query/planner/logical/PlanFragmenter.java

```diff
@@ -89,12 +89,29 @@ private PlanNode process(PlanNode node, Context context) {
 
   @Override
   public void onSubstitution(int receiver, int oldSender, int newSender) {
+    // Change the sender of the receiver to the new sender
     IntList senders = _childPlanFragmentIdsMap.get(receiver);
     senders.rem(oldSender);
     if (!senders.contains(newSender)) {
       senders.add(newSender);
     }
+
+    // Remove the old sender and its children from the plan fragment map
     _planFragmentMap.remove(oldSender);
+
+    IntList pending = new IntArrayList();
```

Comment:
nit: `pending` doesn't seem to descriptive enough IMO - what about something like `stagesToRemove` or `fragmentsToRemove` instead?

Reply:
done

Reply:
I've also changed the code to include the first iteration in the while loop